docs/oidc.md

```diff
@@ -0,0 +1,191 @@
+
+Or you can bundle OIDC authentication with LDAP authorization. In this flow, users authenticate via an IdP and are authorized with the LDAP server.
+
+## Benefits
+
+The use of OIDC and OAuth 2.0 provides the following benefits:
+
+* streamlines authentication and authorization flow, 
+* simplifies user management and configuration: everything is done in a single place
+* enables you to use modern authentication techniques like 2FA, MFA and others supported by IdP
+* improves security as credentials are not sent to nor stored in Percona Server for MongoDB. 
+* reduces cross-application risk - access tokens are granted for specific resources using audience claims. If a token is compromised, the token has a limited lifetime and scope to limit access.
+
+## Configuration 
+
+To configure OIDC /OAuth 2.0 authentication and authorization, you must do the following:
+
+1. Configure external identity provider
+2. Configure authentication in Percona Server for MongoDB
+3. Configure user roles and privileges in Percona Server for MongoDB that will be mapped to IdP groups
+
+This section describes Percona Server for MongoDB configuration for OIDC authentication and the available configuration options. Refer to tutorials for detailed step-by-step instructions:
+
+* [Configure OIDC with Okta](oidc-okta.md)
+* [Configure OIDC with Microsoft Entra](oidc-entra.md)
+* [Configure OIDC with Ping Identity]
+* [Configure OIDC with Keycloak]
+
+### Percona Server for MongoDB configuration options
+
+To configure OIDC authentication in Percona Server for MongoDB, specify the external identity provider configuration for the `oidcIdentityProviders` server parameter. You can set it only at startup. See the [Parameter tuning guide](set-parameter.md) for how to set server parameters.
+
+You can use several IdPs for OIDC authentication. In this case, you must add a configuration for every provider to the `oidcIdentityProviders` server parameter. You must also specify a match pattern for each provider. Usernames are then matched against the match pattern to identify which IdP to authenticate a user with. The order in which IdPs are listed defines their priority. The first IdP that matches the username is used for authentication.
+
+The `oidcIdentityProviders` server parameter contains an array of JSON objects with the following parameters:
+
+| Parameter  | Required    | Description |
+|------------|-------------|-------------| 
+| `issuer`   | Yes         | The URL of the identity provider that the server should accept tokens from. It must be a valid URL that starts with `https://`. |
+| `audience` | Yes         | The audience claim for the identity provider. It is used to verify that the access token is intended for your application. You can specify only one value for the `audience` field. When you use multiple identity providers, the `audience` field must have a unique value. |
```

Comment:
Audience must be unique only for IdPs with the same issuer.
For example the following is valid:
```
{
 issuer: "https://url1",
 audience: "some_audience"
},
{
 issuer: "https://url2",
 audience: "some_audience"
}
```